Packs/MISP/Integrations/MISPV3/MISPV3.py

```diff
@@ -0,0 +1,1411 @@
+            return_error(f'Error in `{demisto.command()}` command: {response}')
+    except PyMISPError:
+        raise DemistoException("Removing the required tag was failed. Please make sure the UUID and tag exist.")
+
+    if is_attribute:
+        response = PYMISP.search(uuid=uuid, controller='attributes')
+        human_readable = f'Tag {tag} has been successfully removed from the attribute {uuid}'
+        return CommandResults(
+            readable_output=human_readable,
+            outputs_prefix='MISP.Attribute',
+            outputs_key_field='ID',
+            outputs=build_attributes_search_response(response),
+            raw_response=response
+        )
+    # event's uuid
+    response = PYMISP.search(uuid=uuid)
+    human_readable = f'Tag {tag} has been successfully removed from the event {uuid}'
+    return CommandResults(
+        readable_output=human_readable,
+        outputs_prefix='MISP.Event',
+        outputs_key_field='ID',
+        outputs=build_events_search_response(response),
+        raw_response=response
+    )
+
+
+def add_sighting(demisto_args: dict):
+    """Adds sighting to MISP attribute
+    """
+    attribute_id = demisto_args.get('id')
+    attribute_uuid = demisto_args.get('uuid')
+    sighting_type = demisto_args['type']  # mandatory arg
+    att_id = attribute_id or attribute_uuid
+    if not att_id:
+        return_error('ID or UUID not specified')
+    sighting_args = {
+        'id': attribute_id,
+        'uuid': attribute_uuid,
+        'type': SIGHTING_TO_TYPE_ID[sighting_type]
+    }
+    sigh_obj = MISPSighting()
+    sigh_obj.from_dict(**sighting_args)
+    response = PYMISP.add_sighting(sigh_obj, att_id)
+    if response.get('message'):
+        return_error(f"An error was occurred: {response.get('message')}")
+    elif response.get('Sighting'):
+        human_readable = f'Sighting \'{sighting_type}\' has been successfully added to attribute {att_id}'
+        return CommandResults(readable_output=human_readable)
+    return_error(f"An error was occurred: {json.dumps(response)}")
+
+
+def test(malicious_tag_ids, suspicious_tag_ids):
+    """
+    Test module.
+    """
+    is_tag_list_valid(malicious_tag_ids)
+    is_tag_list_valid(suspicious_tag_ids)
+    response = PYMISP._prepare_request('GET', 'servers/getPyMISPVersion.json')
+    if PYMISP._check_json_response(response):
+        return 'ok'
+    else:
+        return_error('MISP has not connected.')
+
+
+def add_events_from_feed(demisto_args: dict, use_ssl: bool, proxies: dict):
+    """Gets an OSINT feed from url and publishing them to MISP
+    urls with feeds for example: `https://www.misp-project.org/feeds/`
+    feed format must be MISP.
+    """
+    headers = {'Accept': 'application/json'}
+    url = demisto_args.get('feed')  # type: str
+    url = url[:-1] if url.endswith('/') else url
+    if PREDEFINED_FEEDS.get(url):
+        url = PREDEFINED_FEEDS[url].get('url')  # type: ignore
```

Comment:
these lines + line 1161 should be a helper function called `build_feed_url` or something like that

Reply:
Done